platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/crypto/EnhancedKeyStoreLoader.java

```diff
@@ -1165,4 +1175,340 @@ private interface AddressBookCallback {
+     */
+    private long extractPrivateKeysAndCertsFromPfxFiles(
+            final Map<NodeId, PrivateKey> pfxPrivateKeys, final Map<NodeId, Certificate> pfxCertificates)
+            throws KeyStoreException, KeyLoadingException {
+        final KeyStore legacyPublicStore = resolveLegacyPublicStore();
+        AtomicLong errorCount = new AtomicLong(0);
+
+        iterateAddressBook(addressBook, (i, nodeId, address, nodeAlias) -> {
+            if (isLocal(address)) {
+                // extract private keys for local nodes
+                final Path sPrivateKeyLocation = keyStoreDirectory.resolve("s-private-" + nodeAlias + ".pem");
+                final Path ksLocation = legacyPrivateKeyStore(nodeAlias);
+                if (!Files.exists(sPrivateKeyLocation) && Files.exists(ksLocation)) {
+                    logger.trace(
+                            STARTUP.getMarker(),
+                            "Extracting private signing key for node {} from file {}",
+                            nodeId,
+                            ksLocation.getFileName());
+                    final PrivateKey privateKey =
+                            readLegacyPrivateKey(nodeId, ksLocation, KeyCertPurpose.SIGNING.storeName(nodeAlias));
+                    pfxPrivateKeys.put(nodeId, privateKey);
+                    if (privateKey == null) {
+                        logger.error(
+                                ERROR.getMarker(),
+                                "Failed to extract private signing key for node {} from file {}",
+                                nodeId,
+                                ksLocation.getFileName());
+                        errorCount.incrementAndGet();
+                    } else {
+                        logger.trace(
+                                STARTUP.getMarker(),
+                                "Writing private signing key for node {} to PEM file {}",
+                                nodeId,
+                                sPrivateKeyLocation.getFileName());
+                        try {
+                            writePemFile(true, sPrivateKeyLocation, privateKey.getEncoded());
+                        } catch (IOException e) {
+                            logger.error(
+                                    ERROR.getMarker(),
+                                    "Failed to write private key for node {} to PEM file {}",
+                                    nodeId,
+                                    sPrivateKeyLocation.getFileName());
+                            errorCount.incrementAndGet();
+                        }
+                    }
+                }
+            }
+
+            // extract certificates for all nodes
+            final Path sCertificateLocation = keyStoreDirectory.resolve("s-public-" + nodeAlias + ".pem");
+            Path ksLocation = legacyCertificateStore();
+            if (!Files.exists(sCertificateLocation) && Files.exists(ksLocation)) {
```

Comment:
Some of these new migration methods have very deep nesting, which makes following the code a bit difficult.

Consider following alternative structures:

```
if (boolean) {
    // do nested stuff
}
```

becomes

```
if (!boolean) {
    return;
}

// do stuff
```

and

```
if (boolean) {
    // do stuff
} else {
    // do other stuff
}
```

becomes

```
if (boolean) {
    // do stuff
    return;
}

// do other stuff
```

Reply:
I think this is somewhat arbitrary.  The last job I worked at was a defense contractor which required explicit else statements on all ifs to avoid the ambiguity of whether or not an else statement was left off on purpose or if it was a coding mistake and not considered as a case.    

Another coding philosophy that I have seen enforced at some places is to have a single return statement, which would force explicit casing for code paths.

Reply:
At the end of the day, code with blocks nested 5 deep is very difficult to understand, check for bugs, and maintain. My suggestion is just one way to avoid such deep nesting. Another way would be to make more static helper methods, which can be individually considered.